### ISSUE TITLE
Bulk add/remove references in QueryBuilder

### ISSUE THREAD

**Issue by vmattila** [doctrine/mongodb-odm#1538]
This is somehow related to #1537.

ODM's QueryBuilder supports bulk updates in the modifier operations (http://docs.doctrine-project.org/projects/doctrine-mongodb-odm/en/latest/reference/query-builder-api.html#modifier-operations) and uses their native counterparts in the `flush()` operation.

Adding references to a set of documents similar to `$push` or `$pushArray` directly in QueryBuilder would be nice. Referring to my User/Group example in #1537, we could do something like:

```php
$dm->createQueryBuilder('User')
    ->updateMany()
    ->field('groups')->push($group)
    ->getQuery()
    ->execute();
```

ODM could check the definition of `$groups` association and modify the direct query to match the association settings. For example `@ReferenceMany` may `simple=true` that actually changes the native query behind.

**Comment by malarzm** [comment]
I know we're preparing references when building find via query builder so the query is changed basing on reference type. If that's not done already it should be pretty straightforward to make update query respect reference fields upon encountering them too :+1:

**Comment by malarzm** [comment]
Turns out that the feature request is already implemented, but is broken in version `1.1.2`, works fine before (`1.1.0` and `1.1.1`) according to my coworker's report :) Here's code excerpt from the project:

```
$dm->createQueryBuilder()
            ->update(GenericArtwork::class)
            ->multiple(true)
            ->field('shippings')->addToSet($shipping)
            ->field('id')->in($items)
            ->getQuery()
            ->execute();
```

In 1.1.2 it produces exception because there's `$elemMatch` somehow added in `$addToSet` modificator. Retriaging as bug and assigning to `1.1.3` milestone

**Comment by malarzm** [comment]
#1551 just got merged :)
